taskfile/node.go

```diff
@@ -47,11 +49,15 @@ func NewNode(
 ) (Node, error) {
 	var node Node
 	var err error
-	switch getScheme(entrypoint) {
-	case "http", "https":
+	isGit, err := isGitNode(entrypoint)
+	if err != nil {
+		return nil, err
+	}
+	if isGit {
+		node, err = NewGitNode(entrypoint, dir, insecure, opts...)
+	} else if getScheme(entrypoint) == "http" || getScheme(entrypoint) == "https" {
 		node, err = NewHTTPNode(l, entrypoint, dir, insecure, timeout, opts...)
-	default:
-		// If no other scheme matches, we assume it's a file
+	} else {
 		node, err = NewFileNode(l, entrypoint, dir, opts...)
 	}
```

Comment:
I preferred this as a switch tbh. Could we just update `getScheme` to return "git" if we detect a git URL?

Reply:
Good catch! I totally agree with you!